src/main/java/hudson/plugins/ec2/util/AmazonEC2FactoryImpl.java

```diff
@@ -12,8 +12,10 @@ public class AmazonEC2FactoryImpl implements AmazonEC2Factory {
 
     @Override
     public AmazonEC2 connect(AWSCredentialsProvider credentialsProvider, URL ec2Endpoint) {
-        AmazonEC2 client =
-                new AmazonEC2Client(credentialsProvider, EC2Cloud.createClientConfiguration(ec2Endpoint.getHost()));
+        AmazonEC2 client = AmazonEC2ClientBuilder.standard()
+                .withCredentials(credentialsProvider)
+                .withClientConfiguration(EC2Cloud.createClientConfiguration(ec2Endpoint.getHost()))
+                .build();
```

Comment:
@strangelookingnerd Causes [JENKINS-75390](https://issues.jenkins.io/browse/JENKINS-75390).

Reply:
I’ll look into it and check if there is a possibility that does not require to revert to the deprecated implementation.